### ISSUE TITLE
Clarification on P99 normalization for CODEX protein channels

### ISSUE THREAD

**Issue by JAAY-AI** [lilab-stanford/HEX#13]
Hi, thank you for releasing the HEX code and sample data. I have a question regarding the preprocessing of the CODEX protein labels.

In the paper, the Methods section states that each CODEX biomarker channel was normalized to `[0, 1]` based on the 99th percentile (P99) of its global intensity distribution.

However, in `extract_marker_info_patch.py`, I noticed that the code appears to directly compute the mean intensity of each protein channel within a patch:

```python
patch = c2r.pyramid[0][slc].compute()
mean_intensity = patch.reshape(patch.shape[0], -1).mean(axis=1)
```

I could not find the step where the global P99 value is calculated or where each CODEX channel is divided/clipped by its P99 value.

Could you please clarify:

1. Was P99 normalization already applied to the CODEX images before they were used as input to `extract_marker_info_patch.py`?
2. If so, could you provide the preprocessing procedure or code used to calculate the global P99 for each biomarker channel?
3. Was the P99 calculated across all pixels from all samples/slides for each biomarker, or separately for each sample/slide?
4. Is the provided `sample_data/channel_registered/*.csv` generated from CODEX images that had already undergone this P99 normalization?

This clarification would be very helpful for reproducing the label preprocessing described in the paper and for correctly applying HEX to other spatial molecular datasets.

Thank you very much for your help and for making the code publicly available!

**Comment by zli1893** [comment]
Thanks for pointing this out. The P99 normalization was applied after extract_marker_info_patch.py generated the patch-level marker-expression values. For each marker, the 99th percentile was computed from the patch-level values in the training set, and values were normalized as x / P99 and clipped to [0, 1]. The same training-derived P99 value was then applied to the corresponding validation data; P99 was not computed separately for each slide or sample. This normalization step was separate from extract_marker_info_patch.py.
